aten/src/TH/generic/THTensorEvenMoreMath.cpp

```diff
@@ -176,7 +183,13 @@ void THTensor_(indexCopy)(THTensor *tensor, int dim, THLongTensor *index, THTens
 
     for (i=0; i<numel; i++)
     {
-      THTensor_(select)(tSlice, tensor, dim, index_data[i]);
+      int64_t index_val = index_data[i];
```

Comment:
Per the above, let's not check for duplicates and always error (at least for now)